rust/src/delta.rs

```diff
@@ -60,6 +61,11 @@ pub enum DeltaTableError {
         #[from]
         source: ParquetError,
     },
+    #[error("Failed to convert into Arrow schema: {}", .source)]
```

Comment:
Not a blocker and can be addressed as follow up PRs. I think we might need a more specific error variant here because in the future, other part of the code path could trigger ArrowError as well which is not necessarily schema related, when this happens, the error message could be misleading.